src/python/aqueduct_executor/operators/connectors/data/s3.py

```diff
@@ -38,13 +41,25 @@ def authenticate(self) -> None:
                 % str(e)
             )
 
-        # Check that any user-supplied root directory exists.
-        if self.root_dir != "":
-            # If nothing is returned by this filter call, then the directory does not exist.
-            if len(list(self.s3.Bucket(self.bucket).objects.filter(Prefix=self.root_dir))) == 0:
-                raise Exception(
-                    "Supplied root directory `%s` does not exist in bucket %s."
-                    % (self.root_dir, self.bucket)
+        # Check that any user-supplied root directory exists. If the object key name does not exist,
+        # create a new one.
+        try:
+            if self.root_dir != "":
+                # If nothing is returned by this filter call, then the directory does not exist.
+                if len(list(self.s3.Bucket(self.bucket).objects.filter(Prefix=self.root_dir))) == 0:
+                    if not self.root_dir.endswith("/"):
+                        self.root_dir = self.root_dir + "/"
+                    bucket = self.s3.Bucket(self.bucket)
+                    bucket.put_object(Bucket=self.bucket, Key=self.root_dir)
+        except ClientError as e:
+            status_code = e.response["ResponseMetadata"]["HTTPStatusCode"]
+            if status_code == HTTPStatus.FORBIDDEN:
```

Comment:
interesting, is this condition guaranteed to be a permissions error?

Reply:
Yeah. If you access s3 as a client (as in bot3.client('s3'), then exception classes are very different.
In our case, we are accessing as resource (s3.resource('s3'), when user does not permission, we get 403 (forbidden) http response code.